text/experimental/remote-sampling.md

```diff
@@ -0,0 +1,146 @@
+# Remote Sampling for SDK
+
+This is a proposal to add remotely configurable sampling to OpenTelemetry SDK.
+
+## Motivation
+
+Remote sampling configuration is a useful feature that allows users configuring sampling in a central place (e.g. collector) for
+fleet of running OpenTelemetry SDKs. Apart from a central configuration place the remote sampling offers fine-grained
+per-operation sampling configuration.
+
+Remote sampling is already widely used and adopted in other open-source tracing systems such as [Jaeger](https://www.jaegertracing.io/docs/1.23/sampling/).
+
+## Explanation
+
+This draft is related to [0121-config-service](https://github.com/open-telemetry/oteps/blob/main/text/experimental/0121-config-service.md)
+
+Goals:
+
+* supply configuration from a remote source e.g. OpenTelemetry collector to SDK
+* update configuration in SDK at runtime
+* define remote sampler required configuration
+
+## Internal details
+
+```protobuf
+// Sampling configuration for a single service.
+message ServiceSamplingResponse {
+  // Default sampler when no operation_sampling is matched.
+  Sampler default_sampler = 1;
+  // A list of sampling configurations for a specific operations.
+  // First matching configuration is used.
+  repeated OperationSampling operation_sampling = 2;
+}
+
+// Sampling configuration for an operation.
+// Configuration is used only if all properties (span name, span kind, attributes)
+// are matched. Matching is done at span creation time.
+message OperationSampling {
+  // Sampler for this operation.
+  Sampler sampler = 1;
+  // Span name.
+  string span_name = 2;
+  // Span kind.
+  opentelemetry.proto.trace.v1.Span_SpanKind span_kind = 3;
+  // A list of attributes.
+  opentelemetry.proto.common.v1.KeyValue attributes = 4;
+}
+
+// Sampler configuration. Only build in samplers are supported.
+// see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#built-in-samplers
+message Sampler {
+  // Sampler name, equivalent to OTEL_TRACES_SAMPLER. Example always_on, traceidratio etc.
+  // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration
+  string name = 1; // TODO this can be enum (always_on, always_off, parentbased_traceidratio...)
+  // Sampler argument, equivalent to OTEL_TRACES_SAMPLER_ARG. Example 0.25 for traceidratio sampler.
+  string arg = 2;
+}
+
+// Request to get sampling configuration.
+message RemoteSamplingRequest {
+  string service_name = 1;
+  // TODO shall we add version and/or environment?
```

Comment:
instead of specific things like "version" or "environment" i think arbitrary key/value matching makes sense.

i think this feature would be quite interesting, but unsure how widely used it would be.